sqlmesh/core/macros.py

```diff
@@ -491,6 +491,32 @@ def gateway(self) -> t.Optional[str]:
         """Returns the gateway name."""
         return self.var(c.GATEWAY)
 
+    @property
+    def snapshots(self) -> t.Dict[str, Snapshot]:
+        """Returns the snapshots if available."""
+        return self._snapshots
+
+    @property
+    def this_env(self) -> str:
+        """Returns the name of the current environment in before after all."""
+        if "this_env" not in self.locals:
+            raise SQLMeshError("Environment name is not available in this context")
```

Comment:
Should we instead explain how these are only available in the before/after all stages instead? This message is not particularly helpful for debugging.